docs/explanation/overlays.rst

```diff
@@ -0,0 +1,185 @@
+****************
+``OVERLAY`` Step
+****************
+
+It is sometimes necessary to modify the base filesystem used in when generating
+a parts-based payload. On these occasions, the ``OVERLAY`` step provides the
+means to do so.
+
+Overlay Parameters
+------------------
+
+A part has three parameters that can be used to adjust how the overlay step
+works: ``overlay-packages``, ``overlay-script`` and ``overlay-filter``.
+``overlay-packages`` and ``overlay`` (the filter parameter) behave much the
+same way as the related parameters on the ``STAGE`` step. ``overlay-script``
+likewise behaves similarly to ``override-stage``, including having access to
+the ``partsctl`` command.
+
+An example parts section with overlay parameters looks as follows:
+
+.. code-block:: yaml
+
+    parts:
+      part_with_overlay:
+        plugin: nil
+        overlay_packages:
+          - ed
+        overlay_script: |
+          rm -f ${CRAFT_OVERLAY}/usr/bin/vi ${CRAFT_OVERLAY}/usr/bin/vim*
+          rm -f ${CRAFT_OVERLAY}/usr/bin/emacs*
+          rm -f ${CRAFT_OVERLAY}/bin/nano
+        overlay:
+          - bin
+          - usr/bin
+
+After running this part, the overlay layer (and the final package) will only
+contain ``ed`` as an editor, with vi/vim, emacs, and nano all having been
+removed.
```

Comment:
ed is monospaced while the other editors are not, what is the criteria?

Reply:
The criteria comes from me writing the two halves of the sentence 10 minutes apart. Fixed!